File: raincoat-fabric/src/main/java/dev/uten2c/raincoat/direction/DirectionListener.java
package dev.uten2c.raincoat.direction;

import dev.uten2c.raincoat.Networking;
import dev.uten2c.raincoat.States;
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;

public final class DirectionListener {
    private DirectionListener() {
    }

    public static void register() {
        ClientTickEvents.END_CLIENT_TICK.register(client -> {
            if (States.directionSendRequestedTime == null) {
                return;
            }
            var player = client.player;
            if (player == null) {
                return;
            }
            Networking.sendDirectionUpdate(player.getYaw(), player.getPitch());
        });
    }
}

File: raincoat-fabric/src/main/kotlin/dev/uten2c/raincoat/direction/DirectionListener.kt
package dev.uten2c.raincoat.direction

import dev.uten2c.raincoat.Networking.sendDirectionUpdate
import dev.uten2c.raincoat.States
import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents

object DirectionListener {
    fun register() {
        ClientTickEvents.END_CLIENT_TICK.register listener@{ client ->
            if (States.directionSendRequestedTime == null) {
                return@listener
            }
            val player = client.player ?: return@listener
            sendDirectionUpdate(player.yaw, player.pitch)
        }
    }
}
